Android/src/org/droidplanner/android/fragments/geotag/GeoTagImagesService.java

```diff
@@ -0,0 +1,389 @@
+package org.droidplanner.android.fragments.geotag;
+
+import android.app.Service;
+import android.content.Context;
+import android.content.Intent;
+import android.net.Uri;
+import android.os.AsyncTask;
+import android.os.Build;
+import android.os.Environment;
+import android.os.Handler;
+import android.os.IBinder;
+import android.support.v4.content.LocalBroadcastManager;
+
+import com.o3dr.android.client.utils.data.tlog.TLogParser;
+import com.o3dr.android.client.utils.data.tlog.TLogParserCallback;
+import com.o3dr.android.client.utils.data.tlog.TLogParserFilter;
+import com.o3dr.android.client.utils.geotag.GeoTagAsyncTask;
+
+import org.droidplanner.android.BuildConfig;
+import org.droidplanner.android.utils.Utils;
+import org.droidplanner.android.utils.connection.SshConnection;
+
+import java.io.File;
+import java.io.IOException;
+import java.lang.ref.WeakReference;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.NoSuchElementException;
+
+import timber.log.Timber;
+
+/**
+ * Created by chavi on 10/19/15.
+ */
+public class GeoTagImagesService extends Service {
+    private static final String SSH_USERNAME = "root";
+    private static final String SSH_PASSWORD = "TjSDBkAu";
+    private static final SshConnection soloSshLink = new SshConnection(BuildConfig.SOLO_LINK_IP, SSH_USERNAME, SSH_PASSWORD);
```

Comment:
@chaviw this ssh link won't work if the device has a cellular connection. Take a look at the implementation on Solo to see how to get around that.